packages/framework/src/data/fetcher.ts

```diff
@@ -35,9 +35,25 @@ class EnsembleResponse implements Response {
     response.isLoading = false;
     return response;
   }
+
+  static initiateApi(): EnsembleResponse {
+    const response = new EnsembleResponse();
+    response.statusCode = undefined;
+    response.isLoading = true;
+    return response;
+  }
+
+  static failApi(): EnsembleResponse {
+    const response = new EnsembleResponse();
+    response.statusCode = undefined;
+    response.isLoading = false;
+    return response;
+  }
 }
```

Comment:
Please get rid of these. You can update the constructor to initialize the attributes for new responses. But for failed states, mutate the existing object.